packages/edit-site/src/components/global-styles/screen-revisions/index.js

```diff
@@ -116,10 +117,18 @@ function ScreenRevisions() {
 		!! currentlySelectedRevisionId && ! selectedRevisionMatchesEditorStyles;
 	const shouldShowRevisions = ! isLoading && revisions.length;
 
+	const revisionsCountBadge = () => {
+		if ( revisions.length ) {
```

Comment:
Using `revisions.length` causing a slight flash when updating the number. See around the 13s mark:

https://github.com/WordPress/gutenberg/assets/6458278/81e5c05d-d43d-499d-b97f-1d44a52cdb01

Similar to packages/edit-site/src/components/global-styles/ui.js, perhaps we could rely on `globalStyles?._links?.[ 'version-history' ]?.[ 0 ]?.count`, which comes-preloaded.

The downside is that it tells us the number of revisions only, and doesn't include the unsaved or "reset" items.

Do we want to include these in the revisions count?

If so, we could do the calculation in the `useSelect`, e.g., 

```js
	const { blocks, editorCanvasContainerView, revisionsCount } = useSelect(
		( select ) => {
			const {
				getEntityRecord,
				__experimentalGetCurrentGlobalStylesId,
				__experimentalGetDirtyEntityRecords,
			} = select( coreStore );
			const isDirty = __experimentalGetDirtyEntityRecords().length > 0;
			const globalStylesId = __experimentalGetCurrentGlobalStylesId();
			const globalStyles = globalStylesId
				? getEntityRecord( 'root', 'globalStyles', globalStylesId )
				: undefined;
			let _revisionsCount =
				globalStyles?._links?.[ 'version-history' ]?.[ 0 ]?.count || 0;
			// one for the reset item.
			_revisionsCount++;
			// one for any dirty changes (unsaved).
			if ( isDirty ) {
				_revisionsCount++;
			}
			return {
				editorCanvasContainerView: unlock(
					select( editSiteStore )
				).getEditorCanvasContainerView(),
				blocks: select( blockEditorStore ).getBlocks(),
				revisionsCount: _revisionsCount,
			};
		},
		[]
	);
```

and then render when we have the value:

```jsx
			<ScreenHeader
				title={
					revisionsCount &&
					// translators: %s: number of revisions.
					sprintf( __( 'Revisions (%s)' ), revisionsCount )
				}
```

https://github.com/WordPress/gutenberg/assets/6458278/60294bf8-9e49-4f78-8f39-3740e462780d

-----

On a side note, something I missed is to update the `isResolving` check here to `getRevisions`

https://github.com/WordPress/gutenberg/blob/e4d853010148519279f0ed2d673219fd60f1c9de/packages/edit-site/src/components/global-styles/screen-revisions/use-global-styles-revisions.js#L58-L58

I'll get a PR up.

Reply:
> I'll get a PR up.

- https://github.com/WordPress/gutenberg/pull/56491

Reply:
Makes sense, nice catch on the flash. I will update this to use the pre selected version. Thanks :+1: